app/static/js/slcNewProposal.js

```diff
@@ -86,17 +73,42 @@ function fixStepIndicator(navigateTab) {
 
 // TODO: empty the courseInstructor input after an instructor has been added to the table.
 function callback() {
-  let instructor = $("#courseInstructor").val();
+  let data = $("#courseInstructor").val();
+  data = data.split(",");
+  let instructor = data[0]
+  let username = instructor.split('(').pop().split(')')[0]; // returns 'two'
+  let phone = data[1]
   let tableBody = $("#instructorTable").find("tbody");
   let lastRow = tableBody.find("tr:last");
   let newRow = lastRow.clone();
-  newRow.find("td:eq(0)").text(instructor);
+  newRow.find("td:eq(0) p").text(instructor);
+  newRow.find("td:eq(0) div input").val(phone);
+  newRow.find("td:eq(0) div button").attr("data-id", username);
+  newRow.find("td:eq(0) div input").attr("id", username);
   newRow.prop("hidden", false);
   lastRow.after(newRow);
 }
 
 $("#courseInstructor").on('input', function() {
-  searchUser("courseInstructor", callback);
+  searchUser("courseInstructor", callback, null, true);
+});
+
+$("#instructorTable").on("click", "#instructorPhoneUpdate", function() {
+   var inputId = $(this).attr("data-id")
+   var instructorData = [inputId, $("#" + inputId).val()]
+   $.ajax({
+     url: "/updateInstructorPhone",
+     data: JSON.stringify(instructorData),
+     type: "POST",
+     contentType: "application/json",
+     success: function(response) {
+         msgFlash("Instructor's phone number updated", "success")
+     },
+     error: function(request, status, error) {
+       console.log(status,error);

```

Comment:
You don't want to directly output errors, in general. This could be ok in the javascript, except the controller in this case is passing the exception error straight through. That's too much information exposed. The controller should be sending back a more generic message that couldn't be used to get information about the system. Obviously it's not a huge worry in this application but for general web development it's an important principle. If you really wanted to get all of the error info, the controller could selectively send information based on whether we are in a development environment, or debug = True